api/test/h3-data/h3-risk-map.spec.ts

```diff
@@ -279,27 +284,137 @@ describe('H3 Data Module (e2e) - Risk map', () => {
         });
 
       expect(response.body.errors[0].title).toEqual(
-        `There is no H3 producer data for Material with ID ${material.id} and year 2020`,
+        `There is no H3 Producer data registered for Material with ID ${material.id} for the year 2020 or any year before 2020`,
       );
     });
   });
 
+  test('When I get a calculated H3 Water Risk Map with the necessary input values and there is H3 data, but only for the years after the requested one, I should get a proper error message', async () => {
+    const { material, indicator } = await createWorldForRiskMapGeneration({
+      indicatorType: INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+      fakeTable: 'fakeMaterialTable',
+      fakeColumn: 'fakeMaterialColumn',
+      year: 2010,
+    });
+
+    await h3DataMock({
+      h3TableName: 'fakeIndicatorTable',
+      h3ColumnName: 'fakeIndicatorColumn',
+      additionalH3Data: h3IndicatorExampleDataFixture,
+      indicatorId: indicator.id,
+      year: 2005,
+    });
+
+    jest.spyOn(h3DataRepository, 'getWaterRiskMapByResolution');
```

Comment:
what is this meant to do?

Reply:
Seems that the couple of spies present are not used in the assertions, I guess we can remove them :)

Reply:
it was in the test before, I am not sure I understand 100% what exactly it does here, but I will investigate it

Reply:
I think [ this is the answer](https://github.com/Vizzuality/landgriffon/pull/184#discussion_r765837609) to that question

Reply:
ok, so let's remove them